dev/io.openliberty.org.jboss.resteasy.common.ee10/src/org/jboss/resteasy/plugins/server/servlet/ServletUtil.java

```diff
@@ -0,0 +1,204 @@
+
+      String contentType = request.getContentType();
+      if (contentType != null)
+         headers.getMutableHeaders().putSingle(HttpHeaders.CONTENT_TYPE, contentType);
+
+      Map<String, Cookie> cookies = extractCookies(request);
+      headers.setCookies(cookies);
+
+      // test parsing should throw an exception on error
+      headers.testParsing();
+
+      return headers;
+
+   }
+
+   static Map<String, Cookie> extractCookies(HttpServletRequest request)
+   {
+      Map<String, Cookie> cookies = new HashMap<String, Cookie>();
+      if (request.getCookies() != null)
+      {
+         for (jakarta.servlet.http.Cookie cookie : request.getCookies())
+         {
+            Cookie ck1 = new Cookie.Builder(cookie.getName())
+                    .value(cookie.getValue())
+                    .path(cookie.getPath())
+                    .domain(cookie.getDomain())
+                    .version(cookie.getVersion())
+                    .build();
+            cookies.put(cookie.getName(), ck1);
+
+         }
+      }
+      return cookies;
+   }
+
+   public static List<MediaType> extractAccepts(MultivaluedMap<String, String> requestHeaders)
+   {
+      List<MediaType> acceptableMediaTypes = new ArrayList<MediaType>();
+      List<String> accepts = requestHeaders.get(HttpHeaderNames.ACCEPT);
+      if (accepts == null)
+         return acceptableMediaTypes;
+
+      for (String accept : accepts)
+      {
+         acceptableMediaTypes.addAll(MediaTypeHelper.parseHeader(accept));
+      }
+      return acceptableMediaTypes;
+   }
+
+   public static List<String> extractLanguages(MultivaluedMap<String, String> requestHeaders)
+   {
+      List<String> acceptable = new ArrayList<String>();
+      List<String> accepts = requestHeaders.get(HttpHeaderNames.ACCEPT_LANGUAGE);
+      if (accepts == null)
+         return acceptable;
+
+      for (String accept : accepts)
+      {
+         String[] splits = accept.split(",");
+         for (String split : splits)
+            acceptable.add(split.trim());
+      }
+      return acceptable;
+   }
+
+   public static MultivaluedMap<String, String> extractRequestHeaders(HttpServletRequest request)
+   {
+      Headers<String> requestHeaders = new Headers<String>();
+
+      Enumeration<String> headerNames = request.getHeaderNames();
+      while (headerNames.hasMoreElements())
+      {
+         String headerName = headerNames.nextElement();
+         Enumeration<String> headerValues = request.getHeaders(headerName);
+         while (headerValues.hasMoreElements())
+         {
+            String headerValue = headerValues.nextElement();
+            requestHeaders.add(headerName, headerValue);
+         }
+      }
+      return requestHeaders;
+   }
+
+   public static class InitDataCacheKey {
+       private final String requestUrl;
+       private final String contextPath;
+       
+       public InitDataCacheKey (final String url, final String path) {
```

Comment:
```suggestion
       InitDataCacheKey (final String url, final String path) {
```
No reason for it to be public, but there is reason for it to not be private so making it package protected.